stdlib/src/os/env.mojo

```diff
@@ -51,6 +51,26 @@ fn setenv(name: String, value: String, overwrite: Bool = True) -> Bool:
     return status == 0
 
 
+fn unsetenv(name: String) -> Bool:
+    """Unsets an environment variable.
+
+    Constraints:
+        The function only works on macOS or Linux and returns False otherwise.
+
+    Args:
+        name: The name of the environment variable.
+
+    Returns:
+        True if unsetting the variable succeeded. Otherwise, False is returned.
+    """
+    alias os_is_supported = os_is_linux() or os_is_macos()
+    if not os_is_supported:
+        return False
+
+    var status = external_call["unsetenv", Int32](name.unsafe_ptr())
```

Comment:
```suggestion
    var status = external_call["unsetenv", c_int](name.unsafe_ptr())
```

from `sys.ffi` to be explicit (we'd like to be consistent about doing this with other `external_call`s).

Reply:
Updated!